File: Ontology/ObjectIntersectionOf.cs
using System.Collections.Generic;

namespace Ontology
{
    public class ObjectIntersectionOf: IObjectIntersectionOf
    {
        private readonly IList<IClassExpression> _classExpressions;

        public ObjectIntersectionOf(
            params IClassExpression[] classExpressions
            )
        {
            _classExpressions = classExpressions;
        }

        IList<IClassExpression> IObjectIntersectionOf.ClassExpressions => _classExpressions;

        void IClassExpression.Accept(
            IClassExpressionVisitor visitor
            ) => visitor.Visit(this);

        bool IClassExpression.Evaluate(
            IClassMembershipEvaluator evaluator,
            object                    individual
            ) => evaluator.Evaluate(
                this,
                individual);
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectIntersectionOf.ts
import { IClassExpression } from "./IClassExpression";
import { IClassExpressionVisitor } from "./IClassExpressionVisitor";
import { IClassMembershipEvaluator } from "./IClassMembershipEvaluator";
import { IObjectIntersectionOf } from "./IObjectIntersectionOf";

export class ObjectIntersectionOf implements IObjectIntersectionOf
{
    public constructor(
        public ClassExpressions: IClassExpression[]
        )
    {
    }

    Accept(
        visitor: IClassExpressionVisitor
        )
    {
        visitor.ObjectIntersectionOf(this);
    }
    
    Evaluate(
        evaluator : IClassMembershipEvaluator,
        individual: object
        ): boolean
    {
        return evaluator.ObjectIntersectionOf(
            this,
            individual);
    }
}
